### DISCUSSION TITLE
Access registered Route objects

### DISCUSSION THREAD

**Discussion by axelbdt** [jooby-project/jooby#3082] [Q&A] [answered]
Hi,
Is there an idiomatic way to access a registered route from the handler of another route?
Say I am building an e-commerce app and the Category page must render link to each product. How would you retrieve the Route for Product page and in order to reverse it and render the URL of each product?
Thanks in advance,
I am using the mvc API, with no DI (so far)

**Comment by jknack** [comment, marked as answer]
Multiple ways. Each execution is tied to a route:

```
webRoute(Context ctx) {
   var route = ctx.getRoute();
}
```

You can list all them too:

```
webRoute(Context ctx) {
   var router = ctx.getRouter();
   var routes = router.getRoutes();
}
```

A route is available as soon as you add it to the router:

```
{
      var myRoute = get("/route-api", ctx -> {...})

      // get all the routes registered so far
      var routes = getRoutes();

     // get all the routes
     onStarted(() -> {
           var all = getRoutes();
     });
}
```

**Reply by axelbdt** [reply]
Thank you for the detailed answer.
